stripe/src/main/java/com/stripe/android/view/CardInputView.java

```diff
@@ -40,9 +55,118 @@ private void initView() {
         inflate(getContext(), R.layout.card_input_view, this);
 
         mScrollView = (LockableHorizontalScrollView) findViewById(R.id.root_scroll_view);
-        mCardNumberEditText = (EditText) findViewById(R.id.et_card_number);
+        mCardNumberEditText = (CardNumberEditText) findViewById(R.id.et_card_number);
         mExpiryDateEditText = (EditText) findViewById(R.id.et_expiry_date);
         mCvcNumberEditText = (EditText) findViewById(R.id.et_cvc_number);
-        mCardNumberSpace = (Space) findViewById(R.id.space_in_container);
+        mCardNumberSpace = findViewById(R.id.space_in_container);
+        mCardNumberIsViewed = true;
+
+        mCardNumberEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    scrollLeft();
+                }
+            }
+        });
+
+        mExpiryDateEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
```

Comment:
This one is just plain weird -- pasting conversation here for posterity's sake:
The double call to scrollRight doesn't cause a loop / double action because it's gated in the scrollRight function not to be repeatable. Just the same, why call it twice?
So I took out that first call to scrollRight()... and it doesn't work. I thought it was just in my automated test at first, but sure enough it still doesn't work when i tried it in the simulator.